src/components/CodeEditor/CodeEditor.jsx

```diff
@@ -26,24 +34,73 @@ export default function CodeEditor(props) {
     setChecked(event.target.checked);
   }
 
-  useEffect(() => {
-    checked ? setTheme("vs-dark") : setTheme("vs-light");
-  }, [checked]);
+  function checkError(data) {
+    let errorList = [];
+    let hasError = false;
+    data.errors.forEach((error, index) => {
```

Comment:
```suggestion
    data.errors?.forEach((error, index) => {
```

if there's no compiler error this was throwing an error

Reply:
the question mark at the end of errors is shortcut for saying
```
if (data.errors) {
  data.errors.forEach...
}
```